main.go

```diff
@@ -104,10 +105,29 @@ func (b *bufferJoinLines) Write(p []byte) (n int, err error) {
 
 func singleLinePrint(node ast.Node) string {
 	var buf bufferJoinLines
-	printer.Fprint(&buf, token.NewFileSet(), node)
+	printNode(&buf, token.NewFileSet(), node)
 	return buf.String()
 }
 
+func printNode(w io.Writer, fset *token.FileSet, node ast.Node) {
+	switch node := node.(type) {
+	case exprList:
+		if len(node) == 0 {
+			return
+		}
+		printNode(w, fset, node[0])
+		for _, n := range node[1:] {
+			fmt.Fprintf(w, ", ")
+			printNode(w, fset, n)
+		}
+	default:
+		err := printer.Fprint(w, token.NewFileSet(), node)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "cannot print node %T: %v\n", node, err)
```

Comment:
make `printNode` return an error instead, and we can deal with the error in `singleLinePrint` or its parent

Reply:
(fine to ignore the error in the `printNode` caller for now, as we were doing before)

Reply:
I've changed it to panic when it gets an unsupported node type instead. I don't really see the point of adding an error check to each recursive call when we don't actually care about write errors and if the code is correct we should never get an unsupported node type.

Reply:
Sounds good. I'd rather have a panic for something that should never happen than an untested stderr print.